lib/StreamingStore.ts

```diff
@@ -40,24 +40,68 @@ implements RDF.Source<Q>, RDF.Sink<RDF.Stream<Q>, EventEmitter> {
     }
   }
 
-  protected importToListeners(stream: RDF.Stream<Q>): void {
-    stream.on('data', (quad: Q) => {
-      for (const pendingStream of this.pendingStreams.getPendingStreamsForQuad(quad)) {
-        if (!this.ended) {
-          pendingStream.push(quad);
-          pendingStream.emit('quad', quad);
-        }
+  private onReadable(stream: RDF.Stream<Q>, storeImportStream: PassThrough): void {
+    let streamEnded = false;
+    stream.once('readable', async() => {
+      let quad: Q | null = stream.read();
+      while (quad) {
+        const staticQuad = quad;
+        // eslint-disable-next-line @typescript-eslint/no-loop-func
+        await new Promise<void>(resolve => {
+          const matchStream = this.store.match(
+            staticQuad.subject,
+            staticQuad.predicate,
+            staticQuad.object,
+            staticQuad.graph,
+          );
+
+          matchStream.once('data', () => {
+            matchStream.removeAllListeners();
+            if (streamEnded) {
+              storeImportStream.end();
+            }
+            resolve();
+          });
+
+          matchStream.once('end', () => {
+            if (!this.ended) {
+              storeImportStream.push(staticQuad);
+              for (const pendingStream of this.pendingStreams.getPendingStreamsForQuad(staticQuad)) {
+                pendingStream.push(staticQuad);
+                pendingStream.emit('quad', staticQuad);
+              }
+            }
+            if (streamEnded) {
+              storeImportStream.end();
```

Comment:
Wondering if this is the desired behaviour here.

Let's say our internal store is very slow, and `matchStream` ends very late, later than `stream`.
According to the logic above, late arriving quads will not be considered anymore, while I suspect we still want those slow quads to be considered as quads that do not have to be pushed to the pending streams?